# HG changeset patch
# User hseigel
# Date 1396876974 14400
#      Mon Apr 07 09:22:54 2014 -0400
# Node ID f0f39dbf68aa7fc1ca1332311c783f3a28acd621
# Parent  63574c639b1de9c6198dcff2311e410844922019
8037157: Verify <init> call
Summary: Check for null method
Reviewed-by: coleenp, acorn, mschoene

diff -r 63574c639b1d -r f0f39dbf68aa src/share/vm/classfile/verifier.cpp
--- openjdk/hotspot/src/share/vm/classfile/verifier.cpp	Wed Apr 09 14:07:17 2014 -0700
+++ openjdk/hotspot/src/share/vm/classfile/verifier.cpp	Mon Apr 07 09:22:54 2014 -0400
@@ -2301,6 +2301,11 @@
       methodOop m = instanceKlass::cast(ref_klass)->uncached_lookup_method(
         vmSymbols::object_initializer_name(),
         cp->signature_ref_at(bcs->get_index_u2()));
+      if (m == NULL) {
+        verify_error(ErrorContext::bad_code(bci),
+            "Call to missing <init> method");
+        return;
+      }
       instanceKlassHandle mh(THREAD, m->method_holder());
       if (m->is_protected() && !mh->is_same_class_package(_klass())) {
         bool assignable = current_type().is_assignable_from(
